src/account/AccountStorage.sol

```diff
@@ -79,6 +82,33 @@ function getPermittedCallKey(address addr, bytes4 selector) pure returns (bytes2
 
 using EnumerableSet for EnumerableSet.Bytes32Set;
 
+function toSetValue(FunctionReference functionReference) pure returns (bytes32) {
+    return bytes32(FunctionReference.unwrap(functionReference));
+}
+
+function toFunctionReference(bytes32 setValue) pure returns (FunctionReference) {
+    return FunctionReference.wrap(bytes21(setValue));
+}
+
+// HookData layout:
+// 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF______________________ Hook Function Reference
+// 0x__________________________________________AA____________________ is pre hook
+// 0x____________________________________________BB__________________ is post hook
+
+function toSetValue(HookData memory hookData) pure returns (bytes32) {
+    return bytes32(FunctionReference.unwrap(hookData.hookFunction))
+        | bytes32(hookData.isPreHook ? uint256(1) << 80 : 0) | bytes32(hookData.isPostHook ? uint256(1) << 72 : 0);
+}
+
+function toHookData(bytes32 setValue)
+    pure
+    returns (FunctionReference hookFunction, bool isPreHook, bool isPostHook)
+{
+    hookFunction = FunctionReference.wrap(bytes21(setValue));
+    isPreHook = (uint256(setValue) >> 80) & 0xFF == 1;
+    isPostHook = (uint256(setValue) >> 72) & 0xFF == 1;
```

Comment:
Curious why `0xFF` over `0x01`?

Reply:
I used `0xFF` because I wanted to just compare the least significant byte, and I could have also used `0x01` here. It would behave the same if you only use `toHookData` and `toSetValue` to convert between the types, it could differ if someone manually dirties the upper bits of either of the two bytes though.